src/server/game/Spells/SpellInfo.cpp

```diff
@@ -850,6 +850,8 @@ SpellInfo::SpellInfo(SpellEntry const* spellEntry)
     _auraState = AURA_STATE_NONE;
 
     _allowedMechanicMask = 0;
+
+    _isHeartbeatProcOrPeriodic = (ProcFlags & PROC_FLAG_HEARTBEAT) != 0;
```

Comment:
This seems unneccessary, you could just write `(ProcFlags & PROC_FLAG_HEARTBEAT) != 0` directly inside `SpellInfo::IsHeartbeatProcOrPeriodic`